test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -256,22 +257,18 @@ def test_multiple_res(self):
         script_body_m = "echo 12"
         script_body_l = "echo 20"
 
-        fn_s = self.du.create_temp_file(prefix="PtlPbs_small",
-                                        suffix=".scr",
-                                        body=script_body_s,
-                                        asuser=ROOT_USER)
-        fn_m = self.du.create_temp_file(prefix="PtlPbs_medium",
-                                        suffix=".scr",
-                                        body=script_body_m,
-                                        asuser=ROOT_USER)
-        fn_l = self.du.create_temp_file(prefix="PtlPbs_large",
-                                        suffix=".scr",
-                                        body=script_body_l,
-                                        asuser=ROOT_USER)
-
-        self.du.chmod(path=fn_s, mode=0755, sudo=True)
-        self.du.chmod(path=fn_m, mode=0755, sudo=True)
-        self.du.chmod(path=fn_l, mode=0755, sudo=True)
+        fn_s = self.du.create_dyn_res_script(script_body_s,
+                                             prefix="PtlPbs_small",
+                                             suffix=".scr")
+        fn_m = self.du.create_dyn_res_script(script_body_m,
+                                             prefix="PtlPbs_medium",
+                                             suffix=".scr")
+        fn_l = self.du.create_dyn_res_script(script_body_l,
+                                             prefix="PtlPbs_large",
+                                             suffix=".scr")
+        self.filenames.append(fn_s)
```

Comment:
this might be a bit of work, but it would have been much much cleaner if there was a 'globals' module in PTL, which had a list, say, "file_to_cleanup" defined, which functions like create_dyn_res_script could add filenames to so that at the end of the test tearDown() could check the list and delete the files. This would remove the need for a test writer to remember to clean up the files that they create.

Reply:
There's an existing bug out there for this.  I filed it a few months ago.  If you name your files correctly, PbsTestSuite will clean up for you.  Look at test_runner().  It will clean up after the test